tests/e2e/tests/test_prometheus.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+
+from e2e.utils.prometheus.setup_prometheus_server import *
+
+from e2e.utils.constants import DEFAULT_USER_NAMESPACE
+from e2e.utils.utils import (
+    rand_name,
+)
+from e2e.utils.config import metadata
+
+from e2e.conftest import region
+
+from e2e.fixtures.cluster import cluster # Needed for kustomize.
+from e2e.fixtures.kustomize import (
+    kustomize, # Needed for port_forward.
+    clone_upstream, # Needed for kustomize.
+    configure_manifests, # Needed for kustomize.
+)
+from e2e.fixtures.clients import (
+    kfp_client,
+    port_forward, # Needed for kfp_client.
+    host, # Needed for kfp_client.
+    client_namespace, # Needed for kfp_client.
+    session_cookie, # Needed for kfp_client.
+    login, # Needed for session_cookie.
+    password, # Needed for session_cookie.
+)
+
+from kfp_server_api.exceptions import ApiException as KFPApiException
+
+GENERIC_KUSTOMIZE_MANIFEST_PATH = "../../deployments/vanilla"
+
+@pytest.fixture(scope="class")
+def kustomize_path():
+    return GENERIC_KUSTOMIZE_MANIFEST_PATH
+
+class TestPrometheus:
+    @pytest.fixture(scope="class")
+    def setup(self):
+        metadata_file = metadata.to_file()
+        print(metadata.params)  # These needed to be logged
+        print("Created metadata file for TestPrometheus", metadata_file)
+
+    def test_set_up_prometheus(self, metadata, region, kustomize):
+        global workspace_id
+        cluster_name = metadata.get("cluster_name")
+        workspace_id = set_up_prometheus_for_AMP(cluster_name, region)
+        set_up_prometheus_port_forwarding()
+        check_prometheus_is_running()
```

Comment:
This can be its own test which runs after the setup is complete

e.g.

```
@pytest.fixture
def prometheus_amp(cluster, region, request, metadata, kustomize):
    ...

class TestPrometheus:
    @pytest.fixture(scope="class")
    def setup(self, prometheus_amp):
        set_up_prometheus_port_forwarding()

        metadata_file = metadata.to_file()
        print(metadata.params)  # These needed to be logged
        print("Created metadata file for TestPrometheus", metadata_file)

    def test_set_up_prometheus(self, setup):
        check_prometheus_is_running()
```